app/javascript/angular/dashboard/dropdown-menu.js

```diff
@@ -11,6 +12,15 @@ ManageIQ.angular.app.component('dropdownMenu', {
       vm.dropdown_id = `btn_${vm.widgetId}`;
       vm.buttons = JSON.parse(vm.buttonsData);
     };
+
+    vm.click = function(button) {
+      if (button.refresh) {
```

Comment:
Ideally, dropdown-menu should not deal with anything refresh specific.

I think this should be `if (button.onclick) return button.onclick(); return true;`, and we should be doing some `if (button.refresh) button.onclick = myFunction` logic in widget-wrapper.

But for that, `buttonsData` has to be `<` instead of `@` and we need to do the JSON parsing in the wrapper. (And pass `buttons-data="vm.widgetButtons` without the `{{}}`.